pkg/common/common_test.go

```diff
@@ -411,3 +413,249 @@ func TestMergeMapStringString(t *testing.T) {
 		})
 	}
 }
+
+func TestUnMarshallLimitNamespace(t *testing.T) {
+	testCases := []struct {
+		name           string
+		namespace      string
+		expectedKey    client.ObjectKey
+		expectedDomain string
+		expectedError  bool
+	}{
+		{
+			name:           "when namespace is valid and contains both namespace and domain then return the correct values",
+			namespace:      "exampleNS/exampleGW#domain.com",
+			expectedKey:    client.ObjectKey{Name: "exampleGW", Namespace: "exampleNS"},
+			expectedDomain: "domain.com",
+			expectedError:  false,
+		},
+		{
+			name:           "when namespace is invalid (no '#domain') then return an error",
+			namespace:      "exampleNS/exampleGW",
+			expectedKey:    client.ObjectKey{},
+			expectedDomain: "",
+			expectedError:  true,
+		},
+		{
+			name:           "when namespace contains more than one '#' then return an error",
+			namespace:      "exampleNS/exampleGW#domain.com#extra",
+			expectedKey:    client.ObjectKey{},
+			expectedDomain: "",
+			expectedError:  true,
+		},
+		{
+			name:           "when namespace missing both namespace and gateway parts then return an error",
+			namespace:      "#domain.com",
+			expectedKey:    client.ObjectKey{},
+			expectedDomain: "",
+			expectedError:  true,
+		},
+		{
+			name:           "when namespace has no domain name then return an error",
+			namespace:      "exampleNS/exampleGW#",
+			expectedKey:    client.ObjectKey{},
+			expectedDomain: "",
+			expectedError:  true,
```

Comment:
@eguzki @didierofrivia This is the second test-case that fails. Here I can suggest this chagne to `UnMarshallLimitNamespaceTwoFallingTestCases`, to add a check right after the domain is parsed (see the last lines):
```
func UnMarshallLimitNamespaceTwoFallingTestCases(ns string) (client.ObjectKey, string, error) {
	delimIndex := strings.IndexRune(ns, '#')
	if delimIndex == -1 || strings.ContainsRune(ns[delimIndex+1:], '#') {
		return client.ObjectKey{}, "", fmt.Errorf("failed to split on #")
	}

	gwSplit := ns[:delimIndex]
	domain := ns[delimIndex+1:]

	if domain == "" {
		return client.ObjectKey{}, "", fmt.Errorf("empty domain")
	}
        // rest of the code
```
Here, we'll need to put another error message, I guess? If so, I'll need to refactor tests not only to check the expectancy of the error, but the specific type of it. 

What do you think?

Reply:
That could solve it, beware of the if case that needs to be negated when checking the existence of the rune

Reply:
According to our implementation, this is a valid case, where domain name should be empty and the ObjectKey is a valid struct with correct values.
Thanks, @eguzki and @didierofrivia :angel: